packages/opentelemetry-plugin-document-load/src/utils.ts

```diff
@@ -22,3 +25,42 @@
 export function hasKey<O>(obj: O, key: keyof any): key is keyof O {
   return key in obj;
 }
+
+// '{version}-{traceId}-{spanId}-{TraceFlags}';
+const TRACE_PARENT_REGEX = /^[\da-f]{2}-([\da-f]{32})-([\da-f]{16})-([\da-f]{2})$/;
+
+/**
+ * Extracts information from [traceParent] window property and converts it into {@link types.SpanContext}
+ * @param traceParent - A window property that comes from server.
+ *     It should be dynamically generated server side to have the server's request trace Id,
+ *     a parent span Id that was set on the server's request span,
+ *     and the trace flags to indicate the server's sampling decision
+ *     (01 = sampled, 00 = not sampled).
+ *     for example: '{version}-{traceId}-{spanId}-{sampleDecision}'
+ *     For more information see {@link https://www.w3.org/TR/trace-context/}
+ */
+export function extractServerContext(
```

Comment:
I wonder if this code already exists in a shared package or something, i feel like this code has been written before. (no action here btw)

Reply:
refactored